ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -418,6 +419,27 @@ func (r *RayJobReconciler) releaseComputeResources(ctx context.Context, rayJobIn
 			r.Recorder.Eventf(rayJobInstance, corev1.EventTypeNormal, "Deleted", "Deleted cluster %s", rayJobInstance.Status.RayClusterName)
 		}
 	}
+
+	// Since the name of the Kubernetes Job is the same as the RayJob, we need to set the TTL of the Kubernetes Job to clean
+	// up the Kubernetes Job and its Pods when suspending, and a new submitter Kubernetes Job must be created to resubmit the
+	// Ray job if the RayJob is resumed.
+	if isSuspend {
```

Comment:
We can't use `Spec.Suspend` because it may be set to true when the RayJob is in a `Complete` state, which is not allowed to transition to `Suspended`.